src/Bridge/Doctrine/Orm/Filter/BooleanFilter.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Bridge\Doctrine\Orm\Filter;
+
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Util\QueryNameGenerator;
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Doctrine\ORM\QueryBuilder;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Filter by a boolean value
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ * @author Amrouche Hamza <hamza.simperfit@gmail.com>
+ */
+class BooleanFilter extends AbstractFilter
+{
+    /**
+     * @var string Keyword used to retrieve the value.
+     */
+    private $booleanParameterName;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    public function __construct(ManagerRegistry $managerRegistry, RequestStack $requestStack, string $booleanParameterName, array $properties = null)
+    {
+        parent::__construct($managerRegistry, $properties);
+
+        $this->booleanParameterName = $booleanParameterName;
+        $this->requestStack = $requestStack;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Check whether a value is equal to 1, 0, true, false (case insensitive)
+     *
+     * For each property passed, if the resource does not have such property or if the order value is different from
+     * 1, 0, true, false (case insensitive) the property is ignored.
+     *
+     */
+    public function apply(QueryBuilder $queryBuilder, string $resourceClass, string $operationName = null)
+    {
+        $request = $this->requestStack->getCurrentRequest();
+        if (null === $request) {
+            return;
+        }
+
+        $properties = $this->extractProperties($request);
+
+        foreach ($properties as $property => $boolean) {
+            if (!$this->isPropertyEnabled($property) || !$this->isPropertyMapped($property, $resourceClass)) {
+                continue;
+            }
+
+            if (empty($boolean) && isset($this->properties[$property])) {
+                $boolean = $this->properties[$property];
+            }
+
+            $boolean = strtoupper($boolean);
+         if (!in_array($boolean, ['0', '1','TRUE','FALSE'])) {
```

Comment:
What about empty values ? since filter_var return false on empty strings, do we have to check them with the filter_var or is it ok to return boolean false on empty strings ?